File: opabsl/mlbsl/bool.ml
(*
    Copyright © 2011 MLstate

    This file is part of OPA.

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*)

(* Bool *)
(* import Pervasives *)

(* register-spe-op _and\(&&) : bool -> bool -> bool *)
(* register _and\Pervasives.(&&) : bool -> bool -> bool *)



##register _and : bool, bool -> bool
let _and = ( && )
##register _or : bool, bool -> bool
let _or = ( || )
##register not : bool -> bool
let not = ( not )

##register eq : 'a, 'a -> bool
let eq x y = 0 = ServerLib.compare x y
##register lt : 'a, 'a -> bool
let lt x y = 0 > ServerLib.compare x y
##register gt : 'a, 'a -> bool
let gt x y = 0 < ServerLib.compare x y
##register le : 'a, 'a -> bool
let le x y = 0 >= ServerLib.compare x y
##register ge : 'a, 'a -> bool
let ge x y = 0 <= ServerLib.compare x y


(* register &&\Pervasives.(&&) : bool -> bool -> bool *)
(* (\* let and_bool = ( && ) *\) *)

(* register or_bool : bool -> bool -> bool *)
(* register ||\or_bool : bool -> bool -> bool *)
(* let or_bool = ( || ) *)

(* register not_bool : bool -> bool *)
(* register not_bool : bool -> bool *)
(* let not_bool = ( not ) *)

File: lib/plugins/opabsl/cbsl/bool.c
/*
    Copyright © 2011 MLstate

    This file is part of Opa.

    Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:

    The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
*/
/* bool */


//register jlog : string -> unit
//args(a)
//{
//  UNIT;
//}


##register _and : bool -> bool -> bool
##args(a ,b)
{
  return(a&&b);
}

##register _or : bool -> bool -> bool
##args(a,b)
{
  return(a||b);
}

##register not : bool -> bool 
##args(a)
{
  return(!a);
}

##register eq : 'a -> 'a -> bool
##args(a,b)
{
  return(a==b);
}


##register lt : 'a -> 'a -> bool
##args(a,b)
{
  return(a<b);
}

##register gt : 'a -> 'a -> bool
##args(a,b)
{
  return(a>b);
}

##register le : 'a -> 'a -> bool
##args(a,b)
{
  return(a<=b);
}

##register ge : 'a -> 'a -> bool
##args(a,b)
{
  return(a>=b);
}



/*
(* register &&\Pervasives.(&&) : bool -> bool -> bool *)
(* (\* let and_bool = ( && ) *\) *)

(* register or_bool : bool -> bool -> bool *)
(* register ||\or_bool : bool -> bool -> bool *)
(* let or_bool = ( || ) *)

(* register not_bool : bool -> bool *)
(* register not\not_bool : bool -> bool *)
(* let not_bool = ( not ) *)
*/
